core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -392,7 +392,8 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
 
     configureMockTransfer(Map())
     val iterator = createShuffleBlockIteratorWithDefaults(
-      Map(hostLocalBmId -> toBlockList(hostLocalBlocks.keys, 1L, 1))
+      Map(hostLocalBmId -> toBlockList(hostLocalBlocks.keys, 1L, 1)),
+      blockManager = Some(blockManager)
```

Comment:
`blockManager` is actually the default value of `createShuffleBlockIteratorWithDefaults()`. So it looks like there is no actual difference made by this change.

Reply:
We do change the return of `blockManager.hostLocalDirManager` to `Some(hostLocalDirManager)`, so that the blocks are treated as host-local.

Reply:
I see. That makes sense.

Reply:
Verified locally that the correct code path was reached in `partitionBlocksByFetchMode`. I can add an additional check to verify that `mockExternalBlockStoreClient.getHostLocalDirs` was called in the test.